kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/blockchain/BlockchainSource.java

```diff
@@ -24,30 +25,37 @@
     /**
      * @return the cached balance.
      */
-    int getBalance();
+    BalanceUpdate getBalance();
 
     /**
      * Get balance from network
      * @param callback
      */
-    void getBalance(@NonNull final Callback<Integer> callback);
+    void getBalance(@NonNull final Callback<BalanceUpdate> callback);
 
     /**
-     * Add balance observer in order to listen for updates
+     * Add balance observer in order to start receive balance updates
      * @param observer
      */
-    void addBalanceObserver(@NonNull final Observer<Integer> observer);
+    void addBalanceObserver(@NonNull final Observer<BalanceUpdate> observer);
 
     /**
-     * Remove the balance observer in order to stop listening for balance updates.
+     * Add balance observer and start SSE connection on account balance updates from the blockchain network.
      * @param observer
      */
-    void removeBalanceObserver(@NonNull final Observer<Integer> observer);
+    void addBalanceObserverAndStartListen(@NonNull final Observer<BalanceUpdate> observer);
 
     /**
-     * Create the initiated account trustline with Kin Asset
+     * Remove the balance observer in order to stop receiving balance updates.
+     * @param observer
+     */
+    void removeBalanceObserver(@NonNull final Observer<BalanceUpdate> observer);
```

Comment:
any reason to keep this method? you're using it in the presenter, but also there you probably want to stop listen (using `removeBalanceObserverAndStopListen`) if there are no other listeners, right?

Reply:
removeBalanceObserver is not updating and observers count, so this is the main reason why we need it basically. when we do 'addBalanceObserver' we are not forcing a start connection to the blockchain network if there was a payment and this observer was registered so it will get an update, and this is why we need the removeBalanceObserver because there is nothing to update and no connection live.